udf-compiler/src/test/scala/com/nvidia/spark/OpcodeSuite.scala

```diff
@@ -45,19 +42,1834 @@ class OpcodeSuite extends FunSuite {
+      a
+    }
+    val dataset = List(("a","b","c","d")).toDF("x","y","z","w")
+    val u = makeUdf(myudf)
+    val result = dataset.withColumn("new",u(col("x"),col("y"),col("z"),col("w")))
+    val ref = dataset.withColumn("new",col("x"))
+    checkEquiv(result, ref)
+  }
+
+  test("ALOAD_1") {
+    val myudf: (String,String,String,String) => String = (a,b,c,d) => {
+      b
+    }
+    val dataset = List(("a","b","c","d")).toDF("x","y","z","w")
+    val u = makeUdf(myudf)
+    val result = dataset.withColumn("new",u(col("x"),col("y"),col("z"),col("w")))
+    val ref = dataset.withColumn("new",col("y"))
+    checkEquiv(result, ref)
+  }
+
+  test("ALOAD_2") {
+    val myudf: (String,String,String,String) => String = (a,b,c,d) => {
+      c
+    }
+    val dataset = List(("a","b","c","d")).toDF("x","y","z","w")
+    val u = makeUdf(myudf)
+    val result = dataset.withColumn("new",u(col("x"),col("y"),col("z"),col("w")))
+    val ref = dataset.withColumn("new",col("z"))
+    checkEquiv(result, ref)
+  }
+
+  test("ALOAD_3") {
+    val myudf: (String,String,String,String) => String = (a,b,c,d) => {
+      d
+    }
+    val dataset = List(("a","b","c","d")).toDF("x","y","z","w")
+    val u = makeUdf(myudf)
+    val result = dataset.withColumn("new",u(col("x"),col("y"),col("z"),col("w")))
+    val ref = dataset.withColumn("new",col("w"))
+    checkEquiv(result, ref)
+  }
+
+  test("ASTORE_1,2,3") {
+    val myudf: (String) => String = (a) => {
+      val myString : String = a
+      val myString2 : String = myString
+      val myString3 : String = myString2
+      myString3
+    }
+    val dataset = List("a").toDS()
+    val u = makeUdf(myudf)
+    val result = dataset.withColumn("new",u(col("value")))
+    val ref = dataset.withColumn("new",col("value"))
+    checkEquiv(result, ref)
+  }
+
+  test("FSTORE_1,2,3") {
+    val myudf: (Float) => Float = (a) => {
+      var myFloat : Float = a
+      var myFloat2 : Float = myFloat + a
+      var myFloat3 : Float = myFloat2 + a
+      myFloat3
+    }
+    val dataset = List(5.0f).toDS()
+    val u = makeUdf(myudf)
+    val result = dataset.withColumn("new", u(col("value")))
+    val ref = dataset.withColumn("new",col("value")*3)
+    checkEquiv(result, ref)
+  }
+
+
+  test("LSTORE_2") {
+    val myudf: (Long) => Long = (a) => {
+      var myLong : Long = a
+      myLong
+    }
+    val dataset = List(5L).toDS()
+    val u = makeUdf(myudf)
+    val result = dataset.withColumn("new", u(col("value")))
+    val ref = dataset.withColumn("new",col("value"))
+    checkEquiv(result, ref)
+  }
+
+  test("LSTORE_3") {
+    val myudf: (Int, Long) => Long = (a,b) => {
+      var myLong : Long = b
+      myLong
+    }
+    val dataset = List((1,5L)).toDF("x","y")
+    val u = makeUdf(myudf)
+    val result = dataset.withColumn("new", u(col("x"),col("y")))
+    val ref = dataset.withColumn("new",col("y"))
+    checkEquiv(result, ref)
+  }
+
+  // misc. tests. Boolean check currently failing, can't handle true/false
```

Comment:
relevant?

Reply:
I don't think so.  The "Boolean check" test is passing.
I have removed this comment.